### ISSUE TITLE
Request for architecture document for the HospitalRun

### ISSUE THREAD

**Issue by PRDinesh** [HospitalRun/hospitalrun-frontend#632]
Do we have some architecture document for the HospitalRun.
Like what are the components used and how do they fit together
I got the basic idea that the Frontend runs on Ember and backend on nodejs, but I am looking for some more details to understand the architecture correctly and contribute back meaningfully.

I am not expecting a big document, but a simple write up with few diagrams would suffix.

Basically I am checking on the feasibility to deploy HospitalRun in remote locations in India which has very poor/slow Internet connectivity  (they typically runs in 256kbps and so on)

I want to work /improve the network latency and speed of HospitalRun under low internet speeds

**Comment by tangollama** [comment]
@PRDinesh check out the [following diagram](http://hospitalrun.io/img/HospitalRun_deployment.jpeg). We're recommending a deployment with an appliance-level computer locally, backed-up by a cloud instance. CouchDB gives you syncing "for free." In [CURE International hospitals](cure.org) we're making that experience seamless for the users by using the local facility DNS to route the external domain name to the appliance device in the LAN.

i.e. hospital1.hospitalrun.io is truly on the cloud, but the local DNS says it's this device sitting right here. The CouchDB syncing gives you _basically_ real-time syncing whether in the LAN or outside. Is that helpful?

It's also noteworthy that the system isn't labeled as a 1.0 yet and therefore isn't considered "production-ready." That said, we're using it in some pilot locations in my org and many others are conducting pilots.
